lib/component-normalizer.js

```diff
@@ -61,7 +61,16 @@ module.exports = function normalizeComponent (
     hook = injectStyles
   }
 
-  if (hook) {
+  if (hook && options.functional && injectStyles) {
+    // inject styles for functioal component in vue file
+    const existing = options.render
+    options.render = function renderWithStyleInjection (...args) {
```

Comment:
This piece of code will run in browser without compilation so we need to use ES5